KIP-500.md

```diff
@@ -0,0 +1,131 @@
+[2021-02-26 15:37:11,640] INFO [RaftManager nodeId=1] Completed transition to Unattached(epoch=0, voters=[1], electionTimeoutMs=9037) (org.apache.kafka.raft.QuorumState)
+...
+~~~~
+
+Just like with a ZooKeeper based broker, you can connect to port 9092 (or whatever port you configured) to perform administrative operations or produce or consume data.
+
+~~~~
+$ ./bin/kafka-topics.sh --create --topic foo --partitions 1 --replication-factor 1 --bootstrap-server localhost:9092
+Created topic foo.
+~~~~
+
+# Deployment
+Unlike in ZooKeeper-based mode, where any server can become the controller, in self-managed mode, only a small group of specially selected servers can act as controllers.  The specially selected controller servers will participate in the metadata quorum.  Each KIP-500 controller server is either active, or a hot standby for the current active controller server.
+
+Typically you will select either 3 or 5 servers for this role, depending on the size of your cluster.  Just like with ZooKeeper, you must keep a majority of the controllers alive in order to maintain availability.  So if you have 3 controllers, you can tolerate 1 failure; with 5 controllers, you can tolerate 2 failures.
+
+Each Kafka server now has a new configuration key called `process.roles` which can have the following values:
+
+* If `process.roles` is set to `broker`, the server acts as a self-managed broker.
+* If `process.roles` is set to `controller`, the server acts as a self-managed controller.
+* If `process.roles` is set to `broker,controller`, the server acts as both a self-managed broker and a self-managd controller.
+* If `process.roles` is not set at all then we are assumed to be in ZooKeeper mode.  As mentioned earlier, you can't yet transition back and forth between ZK mode and self-managed mode without reformatting.
+
+Nodes that act as both brokers and controllers are referred to as "combined" nodes.  The advantage of using combined nodes you will have uses fewer Java Virtual Machines (JVMs).  This will allow you to avoid some of the fixed memory overheads associated with JVMs.  The disdavantage is that the controller will be less isolated from the rest of the system.  For example, if activity on the broker causes an out of memory condition, the controller part of the server is not isolated from that OOM condition. 
+
+# Missing Features
+We do not yet support generating or loading KIP-630 metadata snapshots.  This means that after a while, the time required to restart a broker will become very large.  This is a known issue and we are working on implementing snapshots for the next release.
+
+We also don't support any kind of upgrade right now, either to or from self-managed mode.  This is another big gap that we are working on.
+
+Finally, the following Kafka features have not yet been fully implemented:
+
+* Support for security (configuring an Authorizer, setting up SCRAM, delegation tokens, and so forth)
```

Comment:
I technically think it is possible to configure an authorizer (e.g. one that goes against some external datastore like LDAP -- though we haven't tested such a configuration).  But the lack of a tested configuration is perhaps reason to keep your wording as-is.

Maybe append the comment that the security protocols SASL_SSL and SASL_PLAINTEXT are supported for the non-SCRAM mechanisms and the SSL security protocol is also supported, but it is not yet possible to create ACLs for the authenticated identities?

Reply:
We disabled the authorizer APIs though for creating and listing ACLs, etc. So I would call it unsupported.

I will add a note that we support SSL, just to be clear.